firedrake/mesh.py

```diff
@@ -2484,12 +2462,16 @@ def make_vom_from_vom_topology(topology, name):
     # Save vertex reference coordinate (within reference cell) in function
     parent_tdim = topology._parent_mesh.ufl_cell().topological_dimension()
     if parent_tdim > 0:
-        reference_coordinates_fs = functionspace.VectorFunctionSpace(
-            vmesh, "DG", 0, dim=parent_tdim
-        )
-        vmesh.reference_coordinates = dmcommon.fill_reference_coordinates_function(
-            function.Function(reference_coordinates_fs)
-        )
+        reference_coordinates_fs = functionspace.VectorFunctionSpace(topology, "DG", 0, dim=parent_tdim)
+        reference_coordinates_data = dmcommon.reordered_coords(topology.topology_dm, reference_coordinates_fs.dm.getDefaultSection(),
+                                                               (topology.num_vertices(), parent_tdim),
+                                                               reference_coord=True)
+        reference_coordinates = function.CoordinatelessFunction(reference_coordinates_fs,
+                                                                val=reference_coordinates_data,
+                                                                name=_generate_default_mesh_reference_coordinates_name(name))
+        vmesh._reference_coordinates = reference_coordinates
```

Comment:
Why are you creating a `CoordinatelessFunction` here? Also if you are going to attach it to the vertex-only mesh, you should probably rename it to avoid confusion with the `.reference_coordinates` `Function` property.

Reply:
Just to be consistent with how we make coordinates in regular mesh.

Did we have `.reference_coordinates` attribute somewhere else?

Reply:
> Just to be consistent with how we make coordinates in regular mesh.

But `reference_coordinates` are more akin to an attribute of each vertex (i.e. a normal `Function`), rather than a coordinates field.

> Did we have .reference_coordinates attribute somewhere else?

I'm just complaining that it's not descriptive of what it is. If you think it should held onto, then call it `._reference_coordinates_cf` or something.

Reply:
Removed `_reference_coordinates`.